c_src/membrane_vpx_plugin/vpx_decoder.c

```diff
@@ -0,0 +1,150 @@
+#include "vpx_decoder.h"
+
+void handle_destroy_state(UnifexEnv *env, State *state) {
+  UNIFEX_UNUSED(env);
+
+  vpx_codec_destroy(&state->codec_context);
+}
+
+UNIFEX_TERM create(UnifexEnv *env, Codec codec) {
+  UNIFEX_TERM result;
+  State *state = unifex_alloc_state(env);
+
+  switch (codec) {
+  case CODEC_VP8:
+    state->codec_interface = vpx_codec_vp8_dx();
+    break;
+  case CODEC_VP9:
+    state->codec_interface = vpx_codec_vp9_dx();
+    break;
+  }
+
+  if (vpx_codec_dec_init(&state->codec_context, state->codec_interface, NULL,
+                         0)) {
+    result = create_result_error(env, "Failed to initialize decoder");
+    unifex_release_state(env, state);
+    return result;
+  }
+  result = create_result_ok(env, state);
+  unifex_release_state(env, state);
+  return result;
+}
+
+Dimensions get_plane_dimensions(const vpx_image_t *img, int plane) {
+  const int height = (plane > 0 && img->y_chroma_shift > 0)
+                         ? (img->d_h + 1) >> img->y_chroma_shift
+                         : img->d_h;
+
+  int width = (plane > 0 && img->x_chroma_shift > 0)
+                  ? (img->d_w + 1) >> img->x_chroma_shift
+                  : img->d_w;
```

Comment:
Why do you add `+1` to `img->dh` if `plane > 0 && img->y_chroma_shift > 0` ? Apart from that I believe `x_chroma_shift` and `y_chroma_shift` are both specified as `unsigned int` so it's safe to always shift by these values

Reply:
they do it this way in libvpx https://github.com/webmproject/libvpx/blob/253d6365e3f134f4aca6f5fc312336cb501a1c6f/tools_common.c#L210

Reply:
Ok, we can leave it that way ;)